File: setup_cors.py
#!/usr/bin/env python3
"""Configure CORS for Firebase Storage bucket using gcloud auth from Firebase CLI"""

import json
import subprocess
import sys

def set_cors():
    bucket_name = 'mission-board-b8dbc.firebasestorage.app'
    cors_config = [
        {
            "origin": ["*"],
            "method": ["GET"],
            "maxAgeSeconds": 3600
        }
    ]
    
    # Write CORS config to temp file
    with open('cors_config.json', 'w') as f:
        json.dump(cors_config, f)
    
    # Use gsutil via gcloud
    try:
        # First try to activate service account from firebase
        print("Attempting to configure CORS for bucket:", bucket_name)
        
        # Try using firebase's gcloud configuration
        result = subprocess.run([
            'gcloud', 'storage', 'buckets', 'update',
            f'gs://{bucket_name}',
            '--cors-config=cors_config.json'
        ], capture_output=True, text=True)
        
        if result.returncode == 0:
            print("✅ CORS configuration set successfully!")
            print("Your Firebase Storage bucket can now serve images to web clients.")
            return True
        else:
            print("Error:", result.stderr)
            # Try alternate method with gsutil
            print("\nTrying alternate method with gsutil...")
            result = subprocess.run([
                'gsutil', 'cors', 'set', 'cors.json',
                f'gs://{bucket_name}'
            ], capture_output=True, text=True)
            
            if result.returncode == 0:
                print("✅ CORS configuration set successfully!")
                return True
            else:
                print("Error:", result.stderr)
                return False
                
    except FileNotFoundError as e:
        print(f"❌ Error: {e}")
        print("Google Cloud SDK tools not found in PATH")
        return False

if __name__ == '__main__':
    success = set_cors()
    sys.exit(0 if success else 1)

File: setup-cors.js
const { exec } = require('child_process');
const { promisify } = require('util');
const execAsync = promisify(exec);
const https = require('https');

async function getAccessToken() {
  try {
    const { stdout } = await execAsync('gcloud auth application-default print-access-token');
    return stdout.trim();
  } catch (error) {
    console.error('Error getting access token:', error.message);
    return null;
  }
}

async function setCorsViaRest() {
  const bucketName = 'mission-board-b8dbc.firebasestorage.app';
  
  const token = await getAccessToken();
  
  if (!token) {
    console.error('❌ Could not get access token. Make sure gcloud is configured:');
    console.error('  gcloud auth login');
    console.error('  gcloud config set project mission-board-b8dbc');
    return false;
  }

  const corsConfig = {
    cors: [
      {
        origin: ['*'],
        method: ['GET'],
        maxAgeSeconds: 3600
      }
    ]
  };

  const data = JSON.stringify(corsConfig);
  
  const options = {
    hostname: 'storage.googleapis.com',
    path: `/storage/v1/b/${bucketName}?fields=cors`,
    method: 'PATCH',
    headers: {
      'Authorization': `Bearer ${token}`,
      'Content-Type': 'application/json',
      'Content-Length': data.length
    }
  };

  return new Promise((resolve) => {
    const req = https.request(options, (res) => {
      let responseData = '';
      
      res.on('data', (chunk) => {
        responseData += chunk;
      });
      
      res.on('end', () => {
        if (res.statusCode >= 200 && res.statusCode < 300) {
          console.log('✅ CORS configuration set successfully!');
          console.log('Your Firebase Storage bucket can now serve images to any origin.');
          resolve(true);
        } else {
          console.error(`❌ Error (${res.statusCode}):`, responseData);
          resolve(false);
        }
      });
    });

    req.on('error', (error) => {
      console.error('❌ Request error:', error.message);
      resolve(false);
    });

    req.write(data);
    req.end();
  });
}

async function main() {
  const success = await setCorsViaRest();
  if (success) {
    console.log('\n✅ You can now test uploading images in your app!');
  }
  process.exit(success ? 0 : 1);
}

main();
